react-app/src/utils/leaguehelpers.js

```diff
@@ -33,51 +33,107 @@ export default {
 
 	getDate: (league, dateId) => {
 		//dateId: league.scheduledMatches.dateId
-		if(league == null || dateId === ''){
+		var getdate = {};
+
+		if(league === null || dateId === ''){
 			return null;
 		}
 
 		league.dates.forEach((date) => {
 			if(date.id === dateId){
-				return date;
+				getdate = date;
 			}
 		});
+
+		//return leagues.dates.filter((date) => {date.id === dateId});
+
+		return getdate;
 	},
 
 	getNumInLeague: (league, teamId) => {
 		//teamId: league.scheduledMatches.teamOneId
-		if(league === null || dateId === ''){
+		var teamNum = '';
+
+		if(league === null || teamId === ''){
 			return '';
 		}
 
 		league.teams.forEach((team) => {
 			if(team.id === teamId){
-				return team.numInLeague;
+				teamNum = team.numInLeague;
 			}
 		});
 
+		//return leagues.teams.filter((team) => {team.id === teamId});
+
+		return teamNum;
 	},
 
 	getVenueName: (lookups, fieldId) => {
 		//fieldId: league.scheduledMatches.fieldId
+		var venueName;
+
 		if(lookups === null || fieldId === ''){
 			return '';
 		}
 
 		lookups.venues.forEach((venue) => {
 			if(venue.id === fieldId){
-				return venue.name;
+				venueName = venue.name;
 			}
 		});
-		
+
+		return venueName;
 	},
 
 	convertMatchTime: (time) => {
 		//This is just for the schedules bc theyre always at night
+		//we can change it later..
 		let newTime = (parseInt(time) - 1200).toString();
 		return newTime.substr(0,1) +':' + newTime.substr(1,2) + 'pm';
 	},
 
-	
+	getMatchTimes: (league, dateId) => {
+
+		if(league === null || dateId === ''){
+			return null;
+		}
+
+		let matchTimes = [];
+
+		league.scheduledMatches.forEach((match) =>{
```

Comment:
lines 102 through 112 can be replaced with

return league.scheduledMatches.filter((match) => match.dateId === dateId).map((match) => match.matchTime).sort((timeOne, timeTwo) => timeOne - timeTwo);